src/libraries/System.Security.Cryptography.Encoding/tests/Cbor/CborWriter.Tag.cs

```diff
@@ -0,0 +1,108 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Numerics;
+using System.Threading;
+
+namespace System.Security.Cryptography.Encoding.Tests.Cbor
+{
+    internal partial class CborWriter
+    {
+        public void WriteTag(CborTag tag)
+        {
+            WriteUnsignedInteger(CborMajorType.Tag, (ulong)tag);
+            _isTagContext = true;
+        }
+
+        // Additional tagged type support
+
+        internal const string Rfc3339FormatString = "yyyy-MM-ddTHH:mm:ss.FFFFFFFK";
+
+        public void WriteDateTimeOffset(DateTimeOffset value)
+        {
+            string dateString =
+                value.Offset == TimeSpan.Zero ?
+                value.UtcDateTime.ToString(Rfc3339FormatString) : // prefer 'Z' over '+00:00'
+                value.ToString(Rfc3339FormatString);
+
+            WriteTag(CborTag.DateTimeString);
+            WriteTextString(dateString);
+        }
+
+        public void WriteUnixTimeSeconds(long value)
```

Comment:
The reader supports reading fractional seconds, but the writer doesn't support writing them?

I would ask "should this format also accept a DateTimeOffset value since it can do the calculations itself?", but then there's the question of "do you want to include fractional seconds, or not?".  (Or, the simple assertion that "no, you don't"; and callers who do can just manually call `WriteUnixTimeSeconds((value - DateTimeOffset.UnixEpoch).TotalSeconds)`)

Reply:
The reader supports fractional seconds, since they're part of the CBOR spec. I was contemplating potentially making the writer accept a double argument, however in the end I went for making it consistent with `DateTimeOffset.FromUnixTimeSeconds(long)`.

> "should this format also accept a DateTimeOffset value since it can do the calculations itself?"

Probably not, since this would result in loss of information for any value with a nonzero offset. I also considered DateTime, however I find the integer choice to be more straightforward.